proselint/command_line.py

```diff
@@ -1,185 +1,183 @@
+        click.echo(json.dumps(proselint_base, sort_keys=True, indent=4))
+        sys.exit(0)
+
+    if isinstance(config, str):
+        config = Path(config)
+    config = tools.load_options(config)
 
-    config = load_options(config, default)
     if dump_config:
-        print(json.dumps(config, sort_keys=True, indent=4))
-        return
+        click.echo(json.dumps(config, sort_keys=True, indent=4))
+        sys.exit(0)
 
-    if time:
-        # click.echo(timing_test())
-        print("This option does not work for the time being.")
-        return
+    if output_format in Output.names():
+        config["output_format"] = output_format
 
-    # In debug or clean mode, delete cache & *.pyc files before running.
-    if debug or clean:
-        clear_cache()
+    if benchmark:
+        run_benchmark()
+        sys.exit(0)
 
     # Use the demo file by default.
     if demo:
+        log.info("Demo-mode activated")
         paths = [demo_file]
 
-    # Expand the list of directories and files.
-    filepaths = extract_files(list(paths))
-
-    # Lint the files
-    num_errors = 0
-
-    # Use stdin if no paths were specified
-    if len(paths) == 0:
-        filepaths.append('-')
-
-    for fp in filepaths:
-        if fp == '-':
-            fp = '<stdin>'
-            f = sys.stdin
-        else:
-            try:
-                f = click.open_file(fp, 'r', "utf-8", "replace")
-            except Exception:
-                traceback.print_exc()
-                sys.exit(2)
-        errors = lint(f, debug, config)
-        num_errors += len(errors)
-        print_errors(fp, errors, output_json, compact)
-
-    # Return an exit code
-    close_cache_shelves()
-    if num_errors > 0:
-        sys.exit(1)
-    else:
-        sys.exit(0)
-
-
-def extract_files(files):
-    """Expand list of paths to include all text files matching the pattern."""
-    expanded_files = []
-    legal_extensions = [".md", ".txt", ".rtf", ".html", ".tex", ".markdown"]
-
-    for f in files:
-        # If it's a directory, recursively walk through it and find the files.
-        if os.path.isdir(f):
-            for dir_, _, filenames in os.walk(f):
-                for filename in filenames:
-                    fn, file_extension = os.path.splitext(filename)
-                    if file_extension in legal_extensions:
-                        joined_file = os.path.join(dir_, filename)
-                        expanded_files.append(joined_file)
-
-        # Otherwise add the file directly.
-        else:
-            expanded_files.append(f)
-
-    return expanded_files
-
-
-if __name__ == '__main__':
+    # prepare list
+    if paths is None:
+        paths = []
+    if isinstance(paths, str):
+        paths = [Path(paths)]
+    if isinstance(paths, Path):
+        paths = [paths]
+    if isinstance(paths, (list, tuple)):
+        paths = [Path(path) for path in paths]
+    log.debug("Paths to lint: %s", paths)
```

Comment:
```suggestion
    log.debug(f"Paths to lint: {paths}")
```

Reply:
nope, not good, like mentioned earlier this is an optimization to lazy-assemble the string. if debug is never shown, why process the string?